cli: panic within `var put` command with certain combination of inputs

### Nomad version
```
$ nomad version
Nomad v1.7.2-dev
BuildDate 2023-12-08T21:05:05Z
Revision e551814df5f271a3ad2f69ed4fd812ce2807be84
```

### Issue
The locally run Nomad CLI panics with the following output:
```
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/hashicorp/nomad/command.(*VarPutCommand).Run(0x14000638480, {0x140001a8210, 0x2, 0x2})
	github.com/hashicorp/nomad/command/var_put.go:306 +0x1558
github.com/mitchellh/cli.(*CLI).Run(0x14000a4a640)
	github.com/mitchellh/cli@v1.1.5/cli.go:262 +0x4a8
main.Run({0x140001a81f0, 0x4, 0x4})
	github.com/hashicorp/nomad/main.go:110 +0x21c
main.main()
	github.com/hashicorp/nomad/main.go:80 +0x54
```

### Reproduction steps
1. Run Nomad in dev mode via `nomad agent -dev`
2. Create a local file name `spec.nv.hcl` with the following contents:
```
path      = "path/to/variable"
namespace = "default"
```
3. Run the Nomad CLI command `nomad var put @spec.nv.hcl 'key=value'`

--- a/.changelog/19423.txt
+++ b/.changelog/19423.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+cli: Fix a bug in the `var put` command which prevented combining items as CLI arguments and other parameters as flags
+```

--- a/command/var_put.go
+++ b/command/var_put.go
@@ -394,6 +394,13 @@ func (c *VarPutCommand) makeVariable(path string) (*api.Variable, error) {
 		return nil, fmt.Errorf("unknown format flag value")
 	}
 
+	// It is possible a specification file was used which did not declare any
+	// items. Therefore, default the entry to avoid panics and ensure this type
+	// of use is valid.
+	if out.Items == nil {
+		out.Items = make(map[string]string)
+	}
+
 	// Handle cases where values are provided by CLI flags that modify the
 	// the created variable. Typical of a "copy" operation, it is a convenience
 	// to reset the Create and Modify metadata to zero.

--- a/command/var_put_test.go
+++ b/command/var_put_test.go
@@ -137,10 +137,6 @@ func TestVarPutCommand_FlagsWithSpec(t *testing.T) {
 	code := cmd.Run([]string{"-address=" + url, "@" + osFile.Name(), "k1=v1", "k2=v2"})
 	must.Zero(t, code)
 
-	t.Cleanup(func() {
-		_, _ = client.Variables().Delete("test/var", nil)
-	})
-
 	must.StrContains(t, ui.OutputWriter.String(), "path/to/variable")
 	must.StrContains(t, ui.OutputWriter.String(), "\"k1\": \"v1\"")
 	must.StrContains(t, ui.OutputWriter.String(), "\"k2\": \"v2\"")

--- a/command/var_put_test.go
+++ b/command/var_put_test.go
@@ -6,6 +6,7 @@ package command
 import (
 	"encoding/json"
 	"fmt"
+	"os"
 	"regexp"
 	"strings"
 	"testing"
@@ -105,6 +106,44 @@ func TestVarPutCommand_GoodJson(t *testing.T) {
 	require.Equal(t, "default", outVar.Namespace)
 	require.Equal(t, "test/var", outVar.Path)
 	require.Equal(t, api.VariableItems{"k1": "v1", "k2": "v2"}, outVar.Items)
+}
+
+func TestVarPutCommand_FlagsWithSpec(t *testing.T) {
+	ci.Parallel(t)
+
+	// Create a server
+	srv, client, url := testServer(t, true, nil)
+	defer srv.Shutdown()
+
+	ui := cli.NewMockUi()
+	cmd := &VarPutCommand{Meta: Meta{Ui: ui}}
+
+	// Create a temporary file and ensure the proper cleanup is run once the
+	// test ends.
+	osFile, err := os.CreateTemp("", "nomad-cli-var-put-test-*.hcl")
+	must.NoError(t, err)
+
+	t.Cleanup(func() {
+		_ = osFile.Close()
+		_ = os.Remove(osFile.Name())
+	})
+
+	// Write out a partial spec that includes the namespace and variable path.
+	_, err = osFile.Write([]byte("path      = \"path/to/variable\"\nnamespace = \"default\""))
+	must.NoError(t, err)
+
+	// Create the variables, ensure we clean it up, and check the command
+	// response.
+	code := cmd.Run([]string{"-address=" + url, "@" + osFile.Name(), "k1=v1", "k2=v2"})
+	must.Zero(t, code)
+
+	t.Cleanup(func() {
+		_, _ = client.Variables().Delete("test/var", nil)
+	})
+
+	must.StrContains(t, ui.OutputWriter.String(), "path/to/variable")
+	must.StrContains(t, ui.OutputWriter.String(), "\"k1\": \"v1\"")
+	must.StrContains(t, ui.OutputWriter.String(), "\"k2\": \"v2\"")
 }
 
 func TestVarPutCommand_AutocompleteArgs(t *testing.T) {

--- a/command/var_put_test.go
+++ b/command/var_put_test.go
@@ -112,7 +112,7 @@ func TestVarPutCommand_FlagsWithSpec(t *testing.T) {
 	ci.Parallel(t)
 
 	// Create a server
-	srv, client, url := testServer(t, true, nil)
+	srv, _, url := testServer(t, true, nil)
 	defer srv.Shutdown()
 
 	ui := cli.NewMockUi()

